upgrading/glusterfs_upgrades.adoc

```diff
@@ -0,0 +1,100 @@
+[[upgrading-openshift-when-using-glusterfs]]
+= Upgrading {product-title} When Using GlusterFS Container Native Storage
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[special-considerations-for-glusterfs]]
+== Special Considerations for GlusterFS
+When upgrading  {product-title}, you must upgrade the set of nodes where
+GlusterFS pods are running.
+
+
+Special consideration must be taken when upgrading these nodes, as `drain` and
+`unschedule` will not terminate and evacuate the GlusterFS pods because they are
+running as part of a daemonset.
+
+There is also the potential for someone to run an upgrade on multiple nodes at
+the same time, which would lead to data availability issues if more than one was
+hosting GlusterFS pods.
+
+Even if a serial upgrade is running, there is no guarantee sufficient time will
+be given for GlusterFS to complete all of its healing operations before
+GlusterFS on the next node is terminated. This could leave the cluster in a bad
+or unknown state. Therefore, the following procedure is recommended.
+
+[[upgrading-openshift-when-running-glusterfs]]
+== Upgrading {product-title} When Using Containerized GlusterFS
+
+. Upgrade the control plane (the master nodes and etcd nodes).
+
+. Upgrade standard infra nodes (router, registry, logging and metrics).
++
+[NOTE]
+====
+If any of the nodes in those groups are running GlusterFS , perform step 4 of
+this procedure at the same time. GlusterFS nodes must be upgraded along with
+other nodes in their class (`app` versus `infra`), one at a time.
+====
+
+. Upgrade standard nodes running application containers.
++
+[NOTE]
+====
+If any of the nodes in those groups are running GlusterFS , perform step 4 of
+this procedure at the same time. GlusterFS nodes must be upgraded along with
+other nodes in their class (`app` versus `infra`), one at a time.
+====
+
+. Upgrade the {product-title} nodes running GlusterFS one at a time.
+
+.. Run `oc get daemonset` to verify the label found under `NODE-SELECTOR` is:
++
+----
+Default: storagenode=glusterfs
+----
+
+.. Remove the daemonset label from the node:
++
+----
+$ oc label node <node_name> storagenode-
+----
++
+This will cause the GlusterFS pod to terminate on that node.
+
+.. Run the upgrade playbook on the single node where you terminated GlusterFS.
+
+.. When the upgrade completes, relabel the node to be what it was previously:
++
+----
+$ oc label node <node_name> <daemonset_label>
+----
+
+.. Wait for the gluster pod to respawn and appear.
+
+.. `rsh` into the pod and verify all volumes are healed:
++
+----
+$ oc rsh <gluster_pod_name>
+----
+
+.. Run the following command to obtain the volume names:
++
+----
+$ gluster volume list
+----
+
+.. Run the following command on each volume to check the self-heal status:
++
+----
+$ gluster volume heal info
+----
++
+Ensure all of the volumes are healed and there are no outstanding tasks.
```

Comment:
``The heal info command lists all pending entries for a given volume's heal process. A volume is considered healed when `Number of entries` for that volume is 0.``